README.md

```diff
@@ -264,6 +264,37 @@ This option set the limit on how many times the plugin should retry the request,
 By Default: 10
 
 
+---
+### `customRetry`
+  - `handler`. Required
+  - `retries`. Optional
+
+This plugin gives the client an option to pass their own retry callback to handle retries on their own.
+If a `handler` is passed to the `customRetry` object the onus is on the client to invoke the default retry logic in their callback otherwise default cases such as 503 will not be handled
+
+Given example
+```js
+  const customRetryLogic = (req, res, registerDefaultRetry, defaultRetryAfter) => {
+    //If this block is not included all non 500 errors will not be retried
+    if (registerDefaultRetry()){
+      return defaultRetryAfter;
+    }
+
+    //Custom retry logic
+    if (res && res.statusCode === 500 && req.method === 'GET') {
+      return 300
+    }
+    return null
+  }
```

Comment:
I think it might be clearer not to describe this as "registering" another retry handler (both in this PR and in the test code), we're not really adding a handler to an event emitter or something, but just passing an option. 

Also, I think that a custom retry handler needs to be passed the number of attempts so far if it wants to calculate an exponential delay. What do you think about this API instaead

```suggestion
const customRetryLogic = (req, res, attemptCount, getDefaultDelay) => {

    const defaultDelay = getDefaultDelay();
    if (defaultDelay) return defaultDelay();
    
    if (res && res.statusCode === 500 && req.method === 'GET') {
      return 300
    }
    return null
  }
```